src/lib/dns.coffee

```diff
@@ -114,42 +112,27 @@ module.exports = (dnschain) ->
             # At some point we may still want to support this though.
             q = req.question[qIdx=0]
             q.name = q.name.toLowerCase()
-            
+
             ttl = Math.floor(Math.random() * 3600) + 30 # TODO: pick an appropriate TTL value!
             @log.debug "received question", q
 
-            if /\.(bit|p2p)$/.test q.name
-                if S(q.name).endsWith '.bit'
-                    nmcDomain = @namecoinizeDomain q.name
-                    resolver = 'nmc'
-                else
-                    # TODO: bdns-izeDomain
-                    nmcDomain = S(q.name).chompRight('.p2p').s
-                    resolver = 'bdns'
-                
-                @log.debug gLineInfo("resolving via #{resolver}..."), {nmcDomain:nmcDomain, q:q}
-
-                @dnschain[resolver].resolve nmcDomain, (err, result) =>
-                    # @log.warn gLineInfo('blah!'), {result: result}
+            chainTLDs = new RegExp('\\.('+_.map(@dnschain.chains, 'tld').join('|')+')$')
```

Comment:
Can you cache this `chainTLDs` value on `@dnschain.chainTLDsRegex` ?

That way it won't need to be computed on each query.